File: fore-jv/fore-jv-core/src/main/java/co/early/fore/core/utils/text/TextPadder.java
package co.early.fore.core.utils.text;

import static co.early.fore.core.utils.text.TextPadder.Pad.LEFT;

public class TextPadder {

    public enum Pad {
        LEFT,
        RIGHT;
    }

    public static String padText(String textOriginal, int desiredLength, Pad pad, char paddingCharacter) {

        notNull(textOriginal);
        notNull(pad);

        int paddingCharactersRequired = desiredLength - textOriginal.length();

        if (paddingCharactersRequired < 0) {
            throw new IllegalArgumentException("textOriginal is already longer than the desiredLength, textOriginal.length():" + textOriginal.length() + " desiredLength:" + desiredLength);
        }

        StringBuilder sb = new StringBuilder();
        for (int ii = 0; ii < paddingCharactersRequired; ii++) {
            sb.append(paddingCharacter);
        }

        if (pad == LEFT) {
            sb.append(textOriginal);
        } else {
            sb.insert(0, textOriginal);
        }

        return sb.toString();
    }

    private static <T> T notNull(T param) {
        if (param == null) {
            throw new NullPointerException("Parameter must not be null");
        }
        return param;
    }
}

File: fore-kt/fore-kt-core/src/main/java/co/early/fore/core/utils/text/TextPadder.kt
package co.early.fore.core.utils.text

import java.lang.NullPointerException
import java.lang.StringBuilder

object TextPadder {
    fun padText(
        textOriginal: String,
        desiredLength: Int,
        pad: Pad,
        paddingCharacter: Char
    ): String {
        notNull(textOriginal)
        notNull(pad)
        val paddingCharactersRequired = desiredLength - textOriginal.length
        require(paddingCharactersRequired >= 0) { "textOriginal is already longer than the desiredLength, textOriginal.length():" + textOriginal.length + " desiredLength:" + desiredLength }
        val sb = StringBuilder()
        for (ii in 0 until paddingCharactersRequired) {
            sb.append(paddingCharacter)
        }
        if (pad == Pad.LEFT) {
            sb.append(textOriginal)
        } else {
            sb.insert(0, textOriginal)
        }
        return sb.toString()
    }

    private fun <T> notNull(param: T?): T {
        if (param == null) {
            throw NullPointerException("Parameter must not be null")
        }
        return param
    }

    enum class Pad {
        LEFT, RIGHT
    }
}